### PR TITLE
feat: add gupshup provider

### PR DESCRIPTION
**Pull request by davidsoderberg**

### CODE REVIEW

File: providers/gupshup/package.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
+{
+  "name": "@novu/gupshup",
+  "version": "",
```
Comment (scopsy) [inline providers/gupshup/package.json:3]:
```suggestion
  "version": "0.5.1",
```

I think that was causing some of the build errors aswell :)

File: providers/gupshup/src/lib/gupshup.provider.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    const response = await fetch(GupshupSmsProvider.BASE_URL, opts);
+    const body = await response.text();
+    const result = body.split(' | ');
+
+    if (result[0] === 'error') {
+      throw new Error(`${result[1]} ${result[2]}`);
+    }
+
+    return {
+      id: result[2],
```
Comment (scopsy) [inline providers/gupshup/src/lib/gupshup.provider.ts:57]: @davidsoderberg is that a messageId here? Or some object?

**Review by scopsy — APPROVED** [review]
Looks great! My only questions is about their response model to make sure we pass back the actual messageId

**Comment by scopsy** [comment]
@davidsoderberg I think it can be merged right?

**Comment by davidsoderberg** [comment]
> @davidsoderberg I think it can be merged right?

yes did it now 😄

**Pull request merged**